src/cli/update.rs

```diff
@@ -0,0 +1,386 @@
+                for package in packages {
+                    match package {
+                        Package::Conda(p) => {
+                            diff.removed.push(Package::Conda(p));
+                        }
+                        Package::Pypi(p) => {
+                            diff.removed.push(Package::Pypi(p));
+                        }
+                    }
+                }
+                environment_diff.insert(platform, diff);
+            }
+
+            // Remove empty diffs
+            environment_diff.retain(|_, diff| !diff.is_empty());
+
+            result
+                .environment
+                .insert(environment_name.to_string(), environment_diff);
+        }
+
+        // Find environments that were completely removed
+        for (environment_name, environment) in previous
+            .environments()
+            .filter(|(name, _)| !result.environment.contains_key(*name))
+            .collect_vec()
+        {
+            let mut environment_diff = IndexMap::new();
+            for (platform, packages) in environment.packages_by_platform() {
+                let mut diff = PackagesDiff::default();
+                for package in packages {
+                    match package {
+                        Package::Conda(p) => {
+                            diff.removed.push(Package::Conda(p));
+                        }
+                        Package::Pypi(p) => {
+                            diff.removed.push(Package::Pypi(p));
+                        }
+                    }
+                }
+                environment_diff.insert(platform, diff);
+            }
+            result
+                .environment
+                .insert(environment_name.to_string(), environment_diff);
+        }
+
+        // Remove empty environments
+        result.environment.retain(|_, diff| !diff.is_empty());
+
+        result
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.environment.is_empty()
+    }
+
+    // Format the lock-file
+    pub fn print(&self) {
+        enum Change<'i> {
+            Added(&'i Package),
+            Removed(&'i Package),
+            Changed(&'i Package, &'i Package),
+        }
+
+        fn format_package_identifier(package: &Package) -> String {
+            match package {
+                Package::Conda(p) => format!(
+                    "{} {}",
+                    &p.package_record().version.as_str(),
+                    &p.package_record().build
+                ),
+                Package::Pypi(p) => p.data().package.version.to_string(),
+            }
+        }
+
+        for (environment_name, environment) in
+            self.environment.iter().sorted_by(|(a, _), (b, _)| a.cmp(b))
+        {
+            println!(
+                "Environment: {}",
+                consts::ENVIRONMENT_STYLE.apply_to(environment_name),
+            );
+            for (platform, packages) in environment {
```

Comment:
Would it make sense to do a seperate print for packages that change for all platforms? 

e.g.
```
Environment: default
    ~ annotated-types 0.6.0 pyhd8ed1ab_0 -> 0.7.0 pyhd8ed1ab_0
    - importlib-metadata 7.1.0 pyha770c72_0
    + importlib_resources 6.4.0 pyhd8ed1ab_0
    ~ jsonschema 3.2.0 pyhd8ed1ab_3 -> 4.22.0 pyhd8ed1ab_0
    + jsonschema-specifications 2023.12.1 pyhd8ed1ab_0
    ~ openssl 3.3.0 hcfcfb64_0 -> 3.3.0 h2466b09_2
    ...
  Platform: win-64
    ~ vc 14.3 hcf57466_18 -> 14.3 ha32ba9b_20
    ~ vc14_runtime 14.38.33130 h82b7239_18 -> 14.38.33135 h835141b_20
    ~ vs2015_runtime 14.38.33130 hcb4865c_18 -> 14.38.33135 h22015db_20
  Platform: osx-64
    ~ libcxx 16.0.6 hd57cbcb_0 -> 17.0.6 h88467a6_0
  Platform: linux-64
    ~ gcc 12.3.0 h915e2ae_6 -> 12.3.0 h915e2ae_7
    ~ gcc_impl_linux-64 12.3.0 h58ffeeb_6 -> 12.3.0 h58ffeeb_7

Reply:
This would greatly reduce the output which is great for humans, for machines we might want to do the full output with `--json` sometime.

Reply:
This would greatly reduce the output which is great for humans, for machines we might want to do the full output with `--json` sometime.